Routers/BookingRouter.cs

```diff
@@ -0,0 +1,118 @@
+﻿using AirportTicketBooking.Classes;
+using AirportTicketBooking.Repositories;
+using ClassType = TicketClassType.ClassType;
+
+namespace AirportTicketBooking.Routers
+{
+    internal class BookingRouter
+    {
+        private BookingRepositoty _bookingRepositoty;
+        private FlightRouter _flightRouter;
+
+        public BookingRouter()
+        {
+            _bookingRepositoty = new BookingRepositoty();
+        }
+
+        public List<Booking> GetAllData()
+        {
+            return _bookingRepositoty.Load();
+        }
+
+        public List<Booking> GetAll()
+        {
+            return GetAllData().Where(booking => booking.ValidationMessage == null).ToList();
+        }
+
+        public bool TryGetByBookingId(int bookingId, out Booking booking)
+        {
+            booking = GetAll().FirstOrDefault(booking => booking.BookingId == bookingId);
+            return (booking != null);
+        }
+
+        public void Add(Booking booking)
+        {
+            _bookingRepositoty.Save(booking);
+        }
+
+        public void RemoveByBookingId(int bookingId)
+        {
+            if (!TryGetByBookingId(bookingId, out _))
+            {
+                throw new Exception($"Booking with BookingId: {bookingId} doesn't exist");
+            }
+
+            var remainingBookings = GetAll().Where(booking => booking.BookingId != bookingId).ToList();
+
+            _bookingRepositoty.DeleteTheFileData();
+
+            foreach (var booking in remainingBookings)
+            {
+                _bookingRepositoty.Save(booking);
+            }
+        }
+
+        public List<Booking> GetBookingForPassenger(int passengerPassportId)
+        {
+            return GetAll().Where(booking => booking.PassengerPassportId == passengerPassportId).ToList();
+        }
+
+
+        public List<Booking> FilterByPrice(decimal price)
+        {
+            return GetAll().Where(booking => booking.Price == price).ToList();
+        }
```

Comment:
let's do something for these methods (filtration methods)
let's do only one filtration, so instead of calling and using `GetAll()`
let's use the method directly and define a base filter that contains the filtration on validation
```suggestion
        public List<Booking> FilterByPrice(decimal price)
        {
            var bookings = _bookingRepositoty.Load();
            var validBookings = FilterValidBookings(bookings);
            return validBookings.Where(booking => booking.Price == price).ToList();
        }

        private static IEnumerable<Booking> FilterValidBookings(List<Booking> bookings)
        {
            return bookings.Where(booking => string.IsNullOrEmpty(booking.ValidationMessage));
        }
```
it's similar to what you wrote but this uses differed execution which is better for performance
and let's add it to all other methods where we're using `GetAll()`